medusa/server/api/v2/search.py

```diff
@@ -0,0 +1,277 @@
+        if not series:
+            return self._bad_request('Invalid series slug')
+
+        series_obj = Series.find_by_identifier(series)
+        if not series_obj:
+            return self._not_found('Series not found')
+
+        return {
+            'results': collect_episodes_from_search_thread(series_obj)
+        }
+
+    def put(self, identifier):
+        """Queue a backlog search for a range of episodes.
+
+        :param identifier:
+        """
+        if not self.request.body:
+            if identifier == 'daily':
+                return self._search_daily()
+            if identifier == 'backlog':
+                return self._search_backlog()
+            else:
+                return self._bad_request("Body required for search type '{0}'".format(identifier))
+
+        data = json_decode(self.request.body)
+        if identifier == 'backlog':
+            return self._search_backlog(data)
+        if identifier == 'failed':
+            return self._search_failed(data)
+        if identifier == 'manual':
+            return self._search_manual(data)
+
+        return self._bad_request("Invalid search type '{0}'".format(identifier))
+
+    def _search_backlog(self, data=None):
+        """Queue a backlog search for results for the provided episodes.
+
+        :param data:
+        :return:
+        :example:
+            Start a backlog search for show slug tvdb1234 with episodes s01e01, s01e02, s03e03.
+            route: `apiv2/search/backlog`
+            { showSlug: "tvdb1234",
+              episodes: [
+                "s01e01",
+                "s01e02",
+                "s03e03",
+              ],
+              options: {}
+            }
+        """
+        if not data:
+            # Trigger a full backlog search
+            if app.backlog_search_scheduler.forceRun():
+                return self._accepted('Full backlog search started')
+
+            return self._bad_request('Triggering a full backlog search failed')
+
+        if not data.get('showSlug'):
+            return self._bad_request('You need to provide a show slug')
+
+        if not data.get('episodes') and not data.get('season'):
+            return self._bad_request('For a backlog search you need to provide a list of episodes or seasons')
+
+        identifier = SeriesIdentifier.from_slug(data['showSlug'])
+        if not identifier:
+            return self._bad_request('Invalid series slug')
+
+        series = Series.find_by_identifier(identifier)
+        if not series:
+            return self._not_found('Series not found')
+
+        episode_segments = self._get_episode_segments(series, data)
+
+        # If a season is passed, we transform it to a list of episode objects. And merge it with the episode_segment.
+        # This because the backlog search has it's own logic for searching per episode or season packs. And falling back
+        # between them, if configured.
+        if data.get('season') and len(data.get('season')) > 0:
```

Comment:
I'm a bit confused about `len(data.get('season')) > 0`. What is it for?

Reply:
I think it was to check for to have at least one season slug. But in hindsight, I don't think it's required.